File: testcases/benchmark/math/fibonacci.f90
function fibonacci(n) result(res)
    integer, intent(in) :: n
    integer :: res, a, b, temp, i
    if (n <= 1) then
        res = n
        return
    end if
    a = 0
    b = 1
    do i = 2, n
        temp = a + b
        a = b
        b = temp
    end do
    res = b
end function fibonacci

program main
end program main

File: testcases/benchmark/math/fibonacci.rs
fn fibonacci(n: i32) -> i32 {
    if n <= 1 { return n; }
    let mut a = 0;
    let mut b = 1;
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {}
